File: 667_BeautifulArrangemenntII.py
class Solution:
    def constructArray(self, n: int, k: int) -> List[int]:
        output = [1]
        result = 1
        distinc = k
        while distinc != 0:
            output.append(result + distinc)
            result += distinc
            distinc = (distinc-1) * -1 if distinc > 0 else (distinc+1) * -1
        if len(output) == n: return output
        for i in range(k+2,n+1):
            output.append(i)
        return output
    
        # Time Complexity = O(n)
        # Space Complexity = O(n)

File: C++/667_BeautifulArrangemenntII.cpp
class Solution {
public:
    vector<int> constructArray(int n, int k) {
        int result = 1;
        int distinc = k;
        vector<int> output;
        output.push_back(result);
        while (distinc != 0){
            result += distinc;
            output.push_back(result);
            distinc = distinc > 0 ? (distinc - 1) * -1 : (distinc + 1) * -1; 
        }
        if (output.size() == n) return output;
        for (int i = k + 2; i < n+1; i++){
            output.push_back(i);
        }
        return output;
    }
};